doc/language.md

```diff
@@ -801,6 +801,14 @@ this list, but most apps can safely use them:
   As a special exception to this rule, `List(Bool)` may **not** be upgraded to a list of structs,
   because implementing this for bit lists has proven unreasonably expensive.
 
+* A integer field may be promoted to an integer field of larger width while keeping the sign
+* A boolean field may be promoted to an integer field
+* A `Text` field may be promoted to `Data`
```

Comment:
I might be missing something, but a `Text` field includes a NUL-delimiter. When transforming the `Text` into a `Data`, the whole byte-string is preserved exactly as it is, including the `NUL-delimiter` (I'll double check my test now), which means that there is no lost information, right?

Reply:
That's correct. The point is, the NUL terminator is not normally considered to be part of the text itself (e.g. text.size() doesn't include it), but it will be considered part of the data (data.size() _will_ include it), which may be confusing or lead to bugs. But it's not hard to adjust the program code to account for this, in which case the change is acceptable.

Relatedly, if you want old programs to be able to understand messages written by new programs, then the new program has to always append a NUL terminator to the data, so that it can be interpreted as text by old programs.

Reply:
That's a good point. Thanks again for the clarification!